htdocs/api/v0.0.3-dev/Candidates.php

```diff
@@ -115,47 +115,73 @@ public function handlePOST()
+        // siteName as an input to candidate creation
+        $user         = \User::singleton();
+        $centerIDs    = $user->getCenterIDs();
+        $allUserSites = array();
+        $num_sites    = count($centerIDs);
         if ($num_sites == 0) {
             $this->header("HTTP/1.1 401 Unauthorized");
             $this->error("You are not affiliated with any site");
             $this->safeExit(0);
-        }
+        } else if ($num_sites > 1) {
+            foreach ($centerIDs as $key => $centerID) {
+                $center = $this->DB->pselectRow(
+                    "SELECT CenterID as ID, Name FROM psc WHERE CenterID =:cid",
+                    array('cid' => $centerID)
+                );
+                $allUserSites[$centerID] = $center['Name'];
+            }
 
-        if ($num_sites > 1) {
-            $this->header("HTTP/1.1 501 Not Implemented");
-            $this->error(
-                "This API version does not support candidate creation " .
-                "by users with multiple site affiliations. This will be ".
-                "implemented in a future release."
-            );
-            $this->safeExit(0);
-        }
+            $siteName = $data['Candidate']['Site'];
+            $this->verifyField($data, 'Site', $allUserSites);
+            $this->verifyField($data, 'Gender', ['Male', 'Female']);
+            $this->verifyField($data, 'EDC', 'YYYY-MM-DD');
+            $this->verifyField($data, 'DoB', 'YYYY-MM-DD');
+            //Get the CenterID from the provided SiteName
+            $centerID = array_search($siteName, $allUserSites);
 
-        $centerID = $centerIDs[0];
-        $this->verifyField($data, 'Gender', ['Male', 'Female']);
-        $this->verifyField($data, 'DoB', 'YYYY-MM-DD');
-
-        //Candidate::createNew
-        try {
-            $candid = $this->createNew(
-                $centerID,
-                $data['Candidate']['DoB'],
-                $data['Candidate']['EDC'],
-                $data['Candidate']['Gender'],
-                $data['Candidate']['PSCID']
-            );
-
-        } catch(\LorisException $e) {
-            $this->header("HTTP/1.1 500 Internal Server Error");
-            $this->error("Candidate can't be created");
-            $this->safeExit(0);
+            //Candidate::createNew
+            try {
+                $candid = $this->createNew(
+                    $centerID,
+                    $data['Candidate']['DoB'],
+                    $data['Candidate']['EDC'],
+                    $data['Candidate']['Gender'],
+                    $data['Candidate']['PSCID']
+                );
+                $this->header("HTTP/1.1 201 Created");
+                $this->JSON = [
+                               'Meta' => ["CandID" => $candid],
+                              ];
+            } catch(\LorisException $e) {
+                $this->header("HTTP/1.1 400 Bad Request");
+                $this->safeExit(0);
```

Comment:
@driusan i am mostly unfamiliar with the API, is there any reason why we don't pass the exception's error message along?

Reply:
I don't know.. what are the exception messages that might be thrown from createNew? Do they make sense to the end user, or only internal developers? It feels like it's less likely to bite us if we don't blindly pass exception messages to end users in a generic `catch` statement, but that's just my gut feeling.